metadata_writer_nne.c

```diff
@@ -32,27 +32,38 @@
-
-
-struct nne_metadata_descr
-{
-    enum nne_metadata_idx idx;
-    const char *key;
-    int mode_dependent;
-    enum nne_type type;
-    enum iface_event event; // iface event that updates this metadata
-    struct nne_value (*parse_cb)(struct nne_modem *modem, struct md_iface_event *mie);
+static struct nne_value md_iface_parse_mode(struct nne_modem *modem, struct md_iface_event *mie);
+static struct nne_value md_iface_parse_submode(struct nne_modem *modem, struct md_iface_event *mie);
+static struct nne_value md_iface_parse_rssi(struct nne_modem *modem, struct md_iface_event *mie);
+static struct nne_value md_iface_parse_rscp(struct nne_modem *modem, struct md_iface_event *mie);
+static struct nne_value md_iface_parse_ecio(struct nne_modem *modem, struct md_iface_event *mie);
+static struct nne_value md_iface_parse_rsrp(struct nne_modem *modem, struct md_iface_event *mie);
+static struct nne_value md_iface_parse_rsrq(struct nne_modem *modem, struct md_iface_event *mie);
+static struct nne_value md_iface_parse_lac(struct nne_modem *modem, struct md_iface_event *mie);
+static struct nne_value md_iface_parse_cid(struct nne_modem *modem, struct md_iface_event *mie);
+static struct nne_value md_iface_parse_oper(struct nne_modem *modem, struct md_iface_event *mie);
+static struct nne_value md_iface_parse_ipaddr(struct nne_modem *modem, struct md_iface_event *mie);
+static struct nne_value md_iface_parse_dev_state(struct nne_modem *modem, struct md_iface_event *mie);
+
+static struct nne_metadata_descr NNE_METADATA_DESCR[] = {
+    { NNE_IDX_MODE,      "mode",         0, NNE_TYPE_UINT8,  IFACE_EVENT_MODE_CHANGE, md_iface_parse_mode },
+    { NNE_IDX_SUBMODE,   "submode",      0, NNE_TYPE_UINT8,  IFACE_EVENT_MODE_CHANGE, md_iface_parse_submode },
+    { NNE_IDX_RSSI,      "rssi",         1, NNE_TYPE_INT8,   IFACE_EVENT_SIGNAL_CHANGE, md_iface_parse_rssi },
+    { NNE_IDX_RSCP,      "rscp",         1, NNE_TYPE_INT16,  IFACE_EVENT_SIGNAL_CHANGE, md_iface_parse_rscp },
+    { NNE_IDX_ECIO,      "ecio",         1, NNE_TYPE_INT8,   IFACE_EVENT_SIGNAL_CHANGE, md_iface_parse_ecio },
+    { NNE_IDX_RSRP,      "rsrp",         1, NNE_TYPE_INT16,  IFACE_EVENT_SIGNAL_CHANGE, md_iface_parse_rsrp },
+    { NNE_IDX_RSRQ,      "rsrq",         1, NNE_TYPE_INT8,   IFACE_EVENT_SIGNAL_CHANGE, md_iface_parse_rsrq },
+    { NNE_IDX_LAC,       "lac",          0, NNE_TYPE_STRING, IFACE_EVENT_LOC_CHANGE, md_iface_parse_lac },
+    { NNE_IDX_CID,       "cid",          0, NNE_TYPE_STRING, IFACE_EVENT_LOC_CHANGE, md_iface_parse_cid },
+    { NNE_IDX_OPER,      "oper",         0, NNE_TYPE_UINT32, IFACE_EVENT_NW_MCCMNC_CHANGE, md_iface_parse_oper },
+    { NNE_IDX_IPADDR,    "ipaddr",       0, NNE_TYPE_STRING, IFACE_EVENT_IP_ADDR_CHANGE, md_iface_parse_ipaddr },
+    { NNE_IDX_DEV_STATE, "device_state", 0, NNE_TYPE_UINT8,  IFACE_EVENT_DEV_STATE, md_iface_parse_dev_state }
 };
 
+#define NNE_METADATA_DESCR_LEN (sizeof(NNE_METADATA_DESCR) / sizeof(struct nne_metadata_descr))
 
```

Comment:
I guess this need to stay here as it is related to definition of nne_metadata_descr table just above.